### PR TITLE
Add HSA_PATH to hip_Includes in cmake and hipconfig

### PR DESCRIPTION
**Pull request by ansurya**
JIRA issues:
SWDEV-192799
SWDEV-189107

### CODE REVIEW

File: hip-config-hcc.cmake.in [hunk 1]
```diff
@@ -42,8 +54,8 @@ endif()
 set(HIP_COMPILER "@HIP_COMPILER@")
 set(HIP_RUNTIME "@HIP_RUNTIME@")
 
-set_and_check( hip_INCLUDE_DIR "@PACKAGE_INCLUDE_INSTALL_DIR@" )
-set_and_check( hip_INCLUDE_DIRS "${hip_INCLUDE_DIR}" )
+set_and_check( hip_INCLUDE_DIR "@PACKAGE_INCLUDE_INSTALL_DIR@")
```
Comment (emankov) [inline hip-config-hcc.cmake.in]: formatting

File: hip-config-hcc.cmake.in [hunk 2]
```diff
@@ -42,8 +54,8 @@ endif()
...
+set( hip_INCLUDE_DIRS "${hip_INCLUDE_DIR}" "${HSA_PATH}/include" )
```
Comment (emankov) [inline hip-config-hcc.cmake.in]: Leave `set_and_check` at least on `hip_INCLUDE_DIRS`, as long as you suggest that `HSA_PATH` is verified.

File: hip-config-clang.cmake.in [hunk 3]
```diff
@@ -1,4 +1,16 @@
 @PACKAGE_INIT@
+if(NOT DEFINED HSA_PATH)
+    if(NOT DEFINED ENV{HSA_PATH})
+        set(HSA_PATH "/opt/rocm/hsa")
```
Comment (emankov) [inline hip-config-clang.cmake.in]: I'd prefer to add something like `CACHE PATH "Path to which HSA runtime has been installed"` to set command.

File: hip-config-hcc.cmake.in (hunk 2 above)
Comment (ansurya) [reply hip-config-hcc.cmake.in]: Added check for HSA include path

File: hip-config-clang.cmake.in (hunk 3 above)
Comment (ansurya) [reply hip-config-clang.cmake.in]: Added as suggested.

File: hip-config-hcc.cmake.in (hunk 1 above)
Comment (ansurya) [reply hip-config-hcc.cmake.in]: done

**Comment by pfultz2** [comment]
This is completely wrong. `find_path` should be used to find the headers. This will break when the users set `CMAKE_PREFIX_PATH` to hsa header files.

**Comment by mangupta** [comment]
> This is completely wrong. `find_path` should be used to find the headers. This will break when the users set `CMAKE_PREFIX_PATH` to hsa header files.
```sh
$ find /opt/rocm -name *config.cmake
/opt/rocm/hip/lib/cmake/hip/hip-config.cmake
/opt/rocm/lib/cmake/hsakmt/hsakmt-config.cmake
/opt/rocm/lib/cmake/amd_comgr/amd_comgr-config.cmake
/opt/rocm/hcc/lib/cmake/hcc/hcc-config.cmake
```
I don't think hsa headers have a cmake config available. If it did, we could have simply done a find_dependency(hsa). Could you clarify how CMAKE_PREFIX_PATH can find hsa headers without an associated config file?

**Comment by pfultz2** [comment]
> I don't think hsa headers have a cmake config available. If it did, we could have simply done a find_dependency(hsa).

They don't have a cmake config, so you have to manually find them using `find_path`, like how it is done in hcc's config [here](https://github.com/RadeonOpenCompute/hcc/blob/clang_tot_upgrade/lib/hcc-config.cmake.in#L7).

> Could you clarify how CMAKE_PREFIX_PATH can find hsa headers without an associated config file?

The `CMAKE_PREFIX_PATH` is the variable a user sets to tell cmake where the dependencies are installed. However, this requires setting another variable so its broken. If you use cmake's find commands(such as `find_path`, `find_file`, `find_library`, etc) then it will search correctly using the `CMAKE_PREFIX_PATH`.

Ultimately, none of this should be needed. The `hip_INCLUDE_DIR` variables should be used to check for correct paths, but users should use hip's cmake targets such as `hip::host` or `hip::device` instead of these variables. So, transitive dependencies such as hsa's headers shouldn't be added to `hip_INCLUDE_DIR` as the cmake targets(ie `hip::device` and `hip::host`) already add the transitive dependencies.

**Comment by emankov** [comment]
As long as `HSA_PATH` is already used, in order to be compatible with  `CMAKE_PREFIX_PATH`, `HSA_PATH` might be included in `find_path`.

**Pull request merged**